### PR TITLE
refactor: table styling and documenting

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #4089

### CODE REVIEW

File: packages/elements/src/components/table/__styles__/index.tsx [hunk 1]
```diff
@@ -123,9 +271,41 @@ export const ElTableRow = styled.div`
...
+  &:hover,
+  &:focus {
+    box-shadow: 0px 0px 3px var(--color-blue-light);
+
+    .el-table-row {
+      box-shadow: none;
+    }
+  }
+
+  &${elIsActive} {
```
Comment (willmcvay) [inline packages/elements/src/components/table/__styles__/index.tsx:300]: Should that not be `&.${elIsActive}` noting the `.`

File: packages/elements/src/components/table/__styles__/index.tsx [hunk 2]
```diff
@@ -140,6 +320,8 @@ export const ElTableCell = styled.div`
     margin-right: 0;
   }
 
+  ${cellOrders}
+
   ${ElIcon} {
```
Comment (willmcvay) [inline packages/elements/src/components/table/__styles__/index.tsx:325]: I don't think this works... I appreciate, not your add but worth checking

File: packages/elements/src/components/table/index.tsx [hunk 3]
```diff
@@ -25,13 +27,14 @@ const getHeadersFromRows = (rows: Row[]): string[] => {
 export type Cell = {
   label: string
   value: string
-  children: React.ReactNode
+  children?: React.ReactNode
   icon?: IconNames
   statusCircleIntent?: Intent
   cellHasDarkText?: boolean
   narrowTable?: {
```
Comment (willmcvay) [inline packages/elements/src/components/table/index.tsx:34]: Is it worth defining `narrowTable` as it's own type?

File: packages/elements/src/components/table/table.stories.mdx [hunk 4]
```diff
@@ -8,9 +8,16 @@ import {
...
+  TableRowContainer,
 } from './molecules'
 import { Icon } from '../icon'
 import { RenderHtmlMarkup } from '../../helpers/render-html-markup'
+import { Grid, Col, elSpan6, elOffset7 } from '../grid'
+import { Molecule } from '../layout'
+import { Subtitle } from '../typography'
+import { InputGroup } from '../input-group'
+import { Button } from '../button'
+import { StatusIndicator } from '../status-indicator'
```
Comment (willmcvay) [inline packages/elements/src/components/table/table.stories.mdx:20]: A general comment but an `<ArgsTable component={Table} story="Foobar" />` component would be good on this page - one for each of the table cell and table components - just helps with understanding the available props

File: packages/elements/src/components/table/table.stories.mdx [hunk 5]
```diff
@@ -148,6 +185,23 @@ If set, this prop will ensure that in narrow view mode the cell takes up both co
...
 
+### Narrow order 
+
+If you require to change the order of your cells in the mobile/narrow view, you can use the `narrowOrder` property to change the cell's order in the column.
+
+The order can be specified between 1 and 12.
+
+<Canvas>
+  <Story name="TableCell - Narrow order">
+    <TableCell narrowOrder={3}>
```
Comment (willmcvay) [inline packages/elements/src/components/table/table.stories.mdx:196]: How do I do narrowOrder if I am non-React? Just a line of commentry - add this class etc

File: packages/elements/src/components/table/table.stories.mdx [hunk 6]
```diff
@@ -218,6 +272,21 @@ Any other props given are applied to the underlying `div` element, e.g. `onClick
...
 
+#### TableExpandableRowTriggerCell narrowIsFullWidth 
+
+This property will force the trigger cell to be fullwidth of the table on narrow sized tables.
+
+> View in mobile view
+
+<Canvas>
+  <Story name="TableExpandableRowTriggerCell - narrowIsFullWidth">
+    <TableExpandableRowTriggerCell narrowIsFullWidth />
```
Comment (willmcvay) [inline packages/elements/src/components/table/table.stories.mdx:283]: As per above, add this class if you're not using react

File: packages/elements/src/components/table/table.stories.mdx [hunk 7]
```diff
@@ -395,6 +726,7 @@ type Row = {
   expandableContent: React.ReactNode
 }
 ```
+##### Inside cells array
 
 ```
 type Cell = {
```
Comment (willmcvay) [inline packages/elements/src/components/table/table.stories.mdx:732]: If we have an ArgsTable, we can get rid of the inline types in the MD

**Review by willmcvay — CHANGES_REQUESTED** [review]
Just a few small things - nice work though, that table's a pain!

File: packages/elements/src/components/table/__styles__/index.tsx (hunk 2 above)
Comment (bashleigh) [reply packages/elements/src/components/table/__styles__/index.tsx:325]: you mean ElIcon? I'll check it out

File: packages/elements/src/components/table/index.tsx (hunk 3 above)
Comment (bashleigh) [reply packages/elements/src/components/table/index.tsx:34]: can do. Do you know if that argsTable thing works with Interfaces/Types?

File: packages/elements/src/components/table/table.stories.mdx (hunk 4 above)
Comment (bashleigh) [reply packages/elements/src/components/table/table.stories.mdx:20]: sure :)

File: packages/elements/src/components/table/table.stories.mdx (hunk 7 above)
Comment (bashleigh) [reply packages/elements/src/components/table/table.stories.mdx:732]: I couldn't get the argsTable component to work with interfaces or types. If it does work with those then happy days!

File: packages/elements/src/components/table/__styles__/index.tsx (hunk 1 above)
Comment (bashleigh) [reply packages/elements/src/components/table/__styles__/index.tsx:300]: I tried `&.${elIsActive}` but it didn't work. Plus what I wrote is used elsewhere in the file

File: packages/elements/src/components/table/__styles__/index.tsx (hunk 2 above)
Comment (bashleigh) [reply packages/elements/src/components/table/__styles__/index.tsx:325]: Screenshot 2021-06-30 at 16 22 20
this adds a margin right to the icon here

File: packages/elements/src/components/table/table.stories.mdx (hunk 4 above)
Comment (bashleigh) [reply packages/elements/src/components/table/table.stories.mdx:20]: I've added some but if you add an args table for components with no uniquely defined props `export const Example = () => ()` then you get some odd results like the previous component's props will appear in the args table

**Review by willmcvay — APPROVED** [review]

**Pull request merged**